src/worker/exporters/node_exporter.py

```diff
@@ -0,0 +1,161 @@
+        result = 'TimeOut'
+        return result
+    return result.decode()
+
+
+class NodeExporter(BaseExporter):
+    '''Example custom node exporter'''
+
+    def __init__(self, node_fields, exp_config):
+        '''initializes parent class'''
+        super().__init__(node_fields, exp_config)
+
+    def collect(self, field_name):
+        '''Custom collection Method'''
+        value = None
+        if field_name == self.node_fields[0]:
+            cmd = "grep 'eth0' " + self.config['fieldFiles'][field_name]
+            args = shlex.split(cmd)
+            value = shell_cmd(args, 5).split()[1]  # 1 is the column for recv bytes
+            delta = int(value) - int(self.config['counter'][field_name])
+            self.config['counter'][field_name] = value
+            value = delta / (self.config['update_freq'])  # bandwidth
+
+        elif field_name == self.node_fields[1]:
+            cmd = "grep 'eth0' " + self.config['fieldFiles'][field_name]
+            args = shlex.split(cmd)
+            value = shell_cmd(args, 5).split()[9]  # 9 is the column for tx bytes
+            delta = int(value) - int(self.config['counter'][field_name])
+            self.config['counter'][field_name] = value
+            value = delta / (self.config['update_freq'])  # bandwidth
+        else:
+            value = 0
+
+        return value
+
+    def cleanup(self):
+        '''Things that need to be called when signal to exit is given'''
+        logging.info('Received exit signal, shutting down ...')
+
+
+# you will need to initialize your custom metric's file if we are exporting from a file
+# you may also want to initialize the config's counter member for the specific field
+def init_config(job_id, port=None):
+    '''Example of config initialization'''
+    global config
+    if not port:
+        port = 8002
+    config = {
+        'exit': False,
+        'update_freq': 1,
+        'listen_port': port,
+        'publish_interval': 1,
+        'job_id': job_id,
+        'fieldFiles': {},
+        'counter': {}
+    }
+    # initalize field specific config parameters
+    for field_name in FIELD_LIST:
+        if 'net' in field_name:
+            config['fieldFiles'][field_name] = '/proc/net/dev'
+            # initialize counter
+            cmd = "grep 'eth0' " + config['fieldFiles'][field_name]
+            args = shlex.split(cmd)
+            if field_name == 'net_rx':
+                config['counter'][field_name] = shell_cmd(args, 5).split()[1]
+            elif field_name == 'net_tx':
+                config['counter'][field_name] = shell_cmd(args, 5).split()[9]
+
+
+# You can just copy paste this function.
+def init_signal_handler():
+    '''Handles exit signals, User defined signale defined in Base class'''
+    def exit_handler(signalnum, frame):
+        config['exit'] = True
+    signal.signal(signal.SIGINT, exit_handler)
+    signal.signal(signal.SIGTERM, exit_handler)
+
+
+# You can just copy paste this function
```

Comment:
what might be a scenario where user will need to use this function?

Reply:
The get _log_level function is mainly used for the arg parser to help parse log level.